### ISSUE TITLE
Prevent children overriding for AT support.

### ISSUE THREAD

**Issue by simondate** [adaptlearning/adapt_framework#1357]
Adapt currently allows duplication of attributes in both the parent and child. The child will always override the parent.

This is all good in the Framework as it means that you can enable use the parent element to describe the default behaviour and then have granular control of children if you require it by adding the code in there.

However in the Authoring Tool as every single attribute is added automatically and can't be removed.  It's impossible to create any 'default' behaviour. The required behaviour has to be created by the course creator every single time which is really time consuming.

My proposal is that the `_isEnabled` attribute gets reworked. `_isEnabled` should check for a parent that has the same attribute. If there isn't one then it works as it does currently (the extension is enabled).

However it a parent with the attribute is found and with the `_isEnabled` found it should instead change functionality to (enable overriding functionality). If it's true than the child overrides the parent as standard. If it's false the child does not override the parent.

**Comment by oliverfoster** [comment]
did you find a solution @simondate ?

**Comment by simondate** [comment]
No, I don't think it's possible to fix at all. Seems like it would require a huge overhaul of the authoring tool to enable 'opting-out' of applying code that is being applied at parent level
